code/src/string.cpp

```diff
@@ -0,0 +1,35 @@
+#include <cmath>
+#include <string.hpp>
+
+namespace r2d2::string {
+    size_t get_offset_separator(const uint8_t *gps_message, 
+                                const size_t length,
+                                const uint8_t separator) {
+
+        for (size_t i = 0; i < length; i++) {
+            if (gps_message[i] == separator) {
+                return i + 1;
+            }
+        }
+        return 0;
+    }
+
+    int32_t atoi(const uint8_t *string, const size_t length) {
+        int32_t res = 0;
+
+        for (size_t i = 0; i < length; i++) {
+            if (string[i] >= '0' && string[i] <= '9') {
+                res = res * 10 + string[i] - '0';
+            }
+        }
+
+        return res * (length ? (string[0] == '-' ? -1 : 1) : 1);
```

Comment:
Is there a situation where length is actually 0? It would be nicer to add this check as a separate if-statement in the first part of the function, as that is clearer.

Reply:
should be fixed in #8